### PR TITLE
Set up training

### PR DESCRIPTION
**Pull request by MLavrentyev**
Set up the training function and the CNN model so we should be ready to train by the end of this week. Few things I have questions on/still needs to be done after this is merged in:
 - @jeffreyzhu101 need help regarding the file name you're outputting, see line 117 in `main.py`. Please lmk soon
 - need to finish the deciding what to do based on args
 - get a test function going
 - get the lstm model up as well

Since it seems the data preprocessing/fextr is nearly done, lmk if you're able to help out with some of these parts as well. Thanks :)

Also, **don't delete the branch when you merge it in**

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -85,8 +94,145 @@ def recording_file(rec_file):
...
     run.add_argument("saved_model", nargs=1, type=existing_model)
     run.add_argument("--output-file", "-o", nargs="?", default=None, type=str)
-    run.add_argument("recordings", nargs="+", type=recording_file)
+    run.add_argument("recording", nargs="+", type=recording_file)
 
     return parser.parse_args()
 
-# process_audio_directory
+
+def get_data_from_dir(data_dir, preprocess_method):
```
Comment (ArYalavarti) [inline main.py]: Right now I save all extracted features under the directories `./data/processed/<accent>/<accent>-<feature>.npy`, where feature is either `mfcc` or `spectrogram`.

File: main.py [hunk 2]
```diff
@@ -85,8 +94,145 @@ def recording_file(rec_file):
     run = subparsers.add_parser("run", description="Run the model on the given data")
     run.add_argument("saved_model", nargs=1, type=existing_model)
     run.add_argument("--output-file", "-o", nargs="?", default=None, type=str)
-    run.add_argument("recordings", nargs="+", type=recording_file)
+    run.add_argument("recording", nargs="+", type=recording_file)
```
Comment (ArYalavarti) [inline main.py]: What's the recording arg for?

File: main.py [hunk 3]
```diff
@@ -85,8 +94,145 @@ def recording_file(rec_file):
...
+    """
+    Loads the data from a given data directory, giving back both inputs and labels
+    :param data_dir: The directory to load data from
+    :param preprocess_method: The method to use for preprocessing (mfcc | fft)
+    :return: The inputs and labels from the data directory
+    """
+    inputs = None
+    labels = None
+
+    accent_class_folders = [folder for folder in os.listdir(data_dir)
```
Comment (ArYalavarti) [inline main.py]: I'd take a look at the `glob` and `pathlib.Path` libraries. Both have really nice directory parsing functions to make analyzing directories a lot easier.

File: main.py [hunk 4]
```diff
@@ -85,8 +94,145 @@ def recording_file(rec_file):
...
+        class_labels = np.full((class_data.shape[0]), model.accent_classes.index(folder))
+
+        if inputs and labels:
+            inputs = np.concatenate(inputs, class_data)
+            labels = np.concatenate(labels, class_labels)
+        else:
+            inputs = class_data
+            labels = class_labels
+
+    return inputs, labels
```
Comment (ArYalavarti) [inline main.py]: Which part of this code takes into account reading train data vs test data? And is there a format you'd like the files to be named with to read in train vs test?

File: main.py [hunk 5]
```diff
@@ -85,8 +94,145 @@ def recording_file(rec_file):
...
+            grads = tape.gradient(loss, model.trainable_variables)
+            model.optimizer.apply_gradients(zip(grads, model.trainable_variables))
+
+        # Print loss and accuracy
+        epoch_loss = model.loss(train_inputs, train_labels)
+        epoch_acc = model.accuracy(train_inputs, train_labels)
+        print(f"Epoch {e}/{epochs} | Loss: {epoch_loss} | Accuracy: {epoch_acc}")
+
+        # Save the model at the end of the epoch
+        if save_file:
```
Comment (ArYalavarti) [inline main.py]: Right now we're checkpointing at the end of every epoch. Depending on training speed, maybe we should checkpoint every x batches instead.

File: main.py (hunk 2 above)
Comment (MLavrentyev) [reply main.py]: The recording is the recording you're evaluating on (i.e. the .wav file to classify)

File: main.py (hunk 1 above)
Comment (MLavrentyev) [reply main.py]: ok sounds good I think that's what I've been using (might need to change fft to spectrogram though)

File: main.py (hunk 4 above)
Comment (MLavrentyev) [reply main.py]: It's generalizable. The point is that you have a file structure such that it goes
train/
  raw/
    chinese/
    british/
    ...
  processed/
    chinese/
    british/
    ...
test/
  < same thing here>

File: main.py (hunk 5 above)
Comment (MLavrentyev) [reply main.py]: Yeah, I'll look into it if training is super slow

**Pull request merged**